app/home-recipe/RegisterRecipe.tsx

```diff
@@ -43,12 +43,9 @@ export enum RecipeTypes {
     LUNCH = "LUNCH",
     DINNER = "DINNER",
     DESSERT = "DESSERT",
-
 }
-const RegisterRecipe = ({ token, postData, uploadRecipeIcon, propsTrigger, setPropsTrigger,
-    existingFormValue, setExistingFormValue }: RegisterRecipeProps) => {
-    console.log("In Register Recipe: check token", token);
-
+const RegisterRecipe = ({ token, postData, getData, uploadRecipeIcon, propsTrigger, setPropsTrigger,
```

Comment:
I am not prof in nowadays front-end, especially React.
Can we avoid passing token as props? 

It will be the best if we can centralised all the handling logic of token / auth. Each component only handle its own logic and data. 

Let say, registerRecipe in this case, it should only focus on the data of the recipe itself while the user profile / file upload mechanism should be handled by a specify common component / module

Reply:
My thought is something like
**Solution 1:**
```
<context>
    <user module session module>
        <common module util component>
            <React router>
                <Each component page/>
            </React router>
        </common module util component>
    </user module session module>
</context>
```

**Solution 2:**
```
contextHolder = [user module, common module];
<context>
    <user module session module />
    <common module util component />
    <React router>
      <Each component page props=contextHolder/>
    </React router>
</context>
```

Reply:
used localStorage for this